### PR TITLE
[Workspace] Add workspace design documentation

### PR DESCRIPTION
**Pull request by ruanyl**
### Description

Add workspace design documentation

## Testing the changes

- doc: add documentation for workspace

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by ruanyl** [comment]
cc @kavilla @joshuarrrr @AMoo-Miki @zengyan-amazon @ashwin-pc ^^

File: docs/plugins/workspace.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+   - **Features**: Users have the flexibility to configure their Personal Workspace according to their preferences and requirements. 
+   - **Permissions**: Access to the Personal Workspace is limited to the user who owns it, ensuring data privacy and personalization.
+3. Management Workspace
+   - **Purpose**: The Management Workspace is dedicated to administrative tasks and configurations within OpenSearch Dashboards.
+   - **Features**: It encompasses features and tools necessary for system administrators to manage OSD advanced settings, such as
+      index management, security configuration, snapshot management, etc.
+   - **Permissions**: Initially, access to the Management Workspace is set to public during initialization, administrators have the
+      responsibility and privilege to restrict access and define permissions within the Management Workspace.
+
+## Appendix
```
Comment (ashwin-pc) [inline docs/plugins/workspace.md:180]: Is there an issue/proposal for the overall Workspaces feature that we can link to over here? The two links below are useful but only talk about access control and not the overall feature.

**Review by ashwin-pc — COMMENTED** [review]
Firstly, love the clear and objective outline of the workspace plugin and its features. One suggestion i have is to move this doc to the workspaces plugin folder. Usually docs that are specific to just one plugin are kept there since its the most obvious place someone will look to understand the plugin and its usage. This folder is usually reserved for features that span multiple plugins and more abstract docs that dont really make sense elsewhere.

File: docs/plugins/workspace.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
+# Workspace
+Workspace is a pivotal feature within OpenSearch-Dashboards, empowering users to craft distinct workspaces 
+by meticulously curating a selection of features and plugins tailored to specific use cases. This functionality
+not only facilitates clients in the systematic organization of visual assets, including saved_objects such as
+dashboards and visualizations, but also categorizes them logically within dedicated workspaces. This strategic
+approach enhances overall efficiency and manageability, making it an indispensable asset for OpenSearch-Dashboards
+users seeking precision and flexibility in their workflows.
```
[the remark this answers is not in the document]
Comment (ruanyl) [reply docs/plugins/workspace.md]: Sounds good to me, updated accordingly.

File: docs/plugins/workspace.md (hunk 1 above)
Comment (ruanyl) [reply docs/plugins/workspace.md]: I added one link to the feature from product design's perspective, that may help to understand the feature from a high level.

**Review by ashwin-pc — APPROVED** [review]
Nice! LGTM :)

**Review by ashwin-pc — COMMENTED** [review]
Nice, thanks for adding this. Do you plan on adding docs on how to create a new workspace type, register a plugin to be a part of a usecase etc?

Also how does workspaces work with permissions? Really curious to know how workspaces interact with the security plugin and tenants

Approving because this is good as is, but the added details would be really helpful :)

**Review by ashwin-pc — APPROVED** [review]

**Comment by ruanyl** [comment]
> Nice, thanks for adding this. Do you plan on adding docs on how to create a new workspace type, register a plugin to be a part of a usecase etc?
> 
> Also how does workspaces work with permissions? Really curious to know how workspaces interact with the security plugin and tenants
> 
> Approving because this is good as is, but the added details would be really helpful :)

Thank you for the questions! That's really helpful, I will make sure these are well documented.

**Review by SuZhou-Joe — APPROVED** [review]

**Pull request merged**